internal/adapters/operation/operation_storage_redis.go

```diff
@@ -335,12 +317,13 @@ func (r *redisOperationStorage) buildSchedulerHistoryOperationsKey(schedulerName
 	return fmt.Sprintf("operations:%s:lists:history", schedulerName)
 }
 
-func (r *redisOperationStorage) getFinishedOperationsFromHistory(ctx context.Context, schedulerName string, from, to time.Time) (operationsIDs []string, err error) {
-	// TODO(guilhermocc): receive this time range as filter argument
+func (r *redisOperationStorage) getFinishedOperationsFromHistory(ctx context.Context, schedulerName string, page, pageSize int64) (operationsIDs []string, err error) {
 	metrics.RunWithMetrics(operationStorageMetricLabel, func() error {
-		operationsIDs, err = r.client.ZRangeByScore(ctx, r.buildSchedulerHistoryOperationsKey(schedulerName), &redis.ZRangeBy{
-			Min: fmt.Sprint(from.Unix()),
-			Max: fmt.Sprint(to.Unix()),
+		operationsIDs, err = r.client.ZRevRangeByScore(ctx, r.buildSchedulerHistoryOperationsKey(schedulerName), &redis.ZRangeBy{
+			Min:    "-inf",
+			Max:    "+inf",
+			Offset: page * pageSize,
+			Count:  pageSize,
 		}).Result()
 		return err
```

Comment:
Since are you naming the error return as err, you don't need to return it here. In this return statement, I think that the naked return is more appropriate.

Reply:
Hey @arthur29 , unfortunately this return is inside of the function that runs RunWithMetrics and it does not accept the naked return 😢

Reply:
Ok, I understand now, why this error is required.